File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numToIndex;
        for (int i; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (numToIndex.contains(complement)) {
                return {numToIndex[complement], i};
            } else {
                numToIndex[nums[i]] = i;
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/target-sum/submission-0.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        cache = {}
        def memo(i, curTarget):
            if i == len(nums):
                if curTarget == 0:
                    return 1
                else:
                    return 0
            if (i,curTarget) in cache:
                return cache[(i,curTarget)]
            cache[(i,curTarget)] = memo(i + 1, curTarget - nums[i]) + memo(i + 1, curTarget + nums[i])
            return cache[(i,curTarget)]
        return memo(0, target)